test/server/filter_chain_manager_impl_test.cc

```diff
@@ -0,0 +1,146 @@
+#include <chrono>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "envoy/admin/v2alpha/config_dump.pb.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "common/api/os_sys_calls_impl.h"
+#include "common/config/metadata.h"
+#include "common/network/address_impl.h"
+#include "common/network/io_socket_handle_impl.h"
+#include "common/network/listen_socket_impl.h"
+#include "common/network/socket_option_impl.h"
+#include "common/network/utility.h"
+#include "common/protobuf/protobuf.h"
+
+#include "server/configuration_impl.h"
+#include "server/filter_chain_manager_impl.h"
+#include "server/listener_manager_impl.h"
+
+#include "extensions/filters/listener/original_dst/original_dst.h"
+#include "extensions/transport_sockets/tls/ssl_socket.h"
+
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/server/utility.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/threadsafe_singleton_injector.h"
+#include "test/test_common/utility.h"
+
+#include "absl/strings/escaping.h"
+#include "absl/strings/match.h"
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::InSequence;
+using testing::Invoke;
+using testing::NiceMock;
+using testing::Return;
+using testing::ReturnRef;
+using testing::Throw;
+
+namespace Envoy {
+namespace Server {
+
+class MockFilterChainFactoryBuilder : public FilterChainFactoryBuilder {
+  std::unique_ptr<Network::FilterChain>
+  buildFilterChain(const ::envoy::api::v2::listener::FilterChain&) const override {
+    // A place holder to be found
```

Comment:
Hm? Could you elaborate on this and/or add a `TODO`?

Reply:
Will add comment:
The test doesn't use the value of the returned ptr but it cannot be nullptr.